### DISCUSSION TITLE
Proper Method for Testing Sharding

### DISCUSSION THREAD

**Discussion by Plinpod** [discordjs/discord.js#9721] [Q&A] [answered]
Hello,

I was wondering if there was a recommended method for testing sharding in Discord.JS

My bot has reached the 2,500 server limit, and I'm trying to implement sharding. I believe I have everything set up properly via the documentation, but I don't know how to test the changes in my environment. So is it possible to shard a test version of a bot across just two servers? 

Thanks!

**Comment by kyranet** [comment, marked as answer]
If your bot isn't very heavy and is running fine, you can use internal sharding instead, which often requires zero changes on your end, see [the tip in the guide](https://discordjs.guide/sharding/):

![image](https://github.com/discordjs/discord.js/assets/24852502/4ba6f0cd-bef6-484c-a5a4-ddbd851b3465)

I recommend using internal sharding over process sharding (`ShardingManager`) mainly on the basis of efficiency, as it uses a lot less memory to run, and simplicity, as you don't have to implement code specifically for it, specially if you do cross-shard tasks.

**Reply by Plinpod** [reply]
Thanks @kyranet  

That's what I did to keep the bot operational but I wasn't sure if that was best practice. 

I do have some bot code that could benefit from process sharding, but maybe it's better to handle concurrency within my code itself and then just use internal sharding on the Discord side.
